cl/sentinel/discovery.go

```diff
@@ -90,6 +98,9 @@ func (s *Sentinel) listenForPeers() {
 		log.Warn("Could not connect to static peers", "reason", err)
 	}
 
+	// limit the number of goroutines opening connection with peers
+	maxPeerOpening := make(chan struct{}, GoRoutinesOpeningPeerConnections)
```

Comment:
never close?

Reply:
Now using semaphore, but what do you mean close here?

Reply:
As it's now listening to `ctx.Done()` we shouldn't need any close.

Reply:
Not closing this one, you close please if it's all good

Reply:
> As it's now listening to `ctx.Done()` we shouldn't need any close.

By close i mean `close(maxPeerOpening)`. But it's fine to not close long-living - which garbage-collectable (no global pointer on it): https://groups.google.com/g/golang-nuts/c/pZwdYRGxCIk/m/rGr1D-uTQMEJ